.github/workflows/publish.yaml

```diff
@@ -20,3 +20,26 @@ jobs:
       - run: npm publish
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_RELEASE_KEY }}
+
+  build_and_push_docker_image:
+    name: Push Docker image to Docker Hub
+    if: github.repository_owner == 'NethermindEth'
```

Comment:
Why do we need the condition?

Reply:
I would assume, so that forks don't try to publish?

Reply:
I think the point is that their CI doesn't even try to do that. If it does, it will result in extra-work, a failure and paint CI red. So, we're just making life a bit less confusing for contributors. At least, that is the reason for why I did stuff like that before. In that case, let's wait for Carmen's response

Reply:
Yes, I think @murcake explained perfectly, even if this workflow won't work for other repos because of the secrets, it shouldn't run for them at all.
When I implemented this workflow I also had a look into [nethermind](https://github.com/NethermindEth/nethermind/blob/master/.github/workflows/build-nethermind-docker-images.yml) and [juno](https://github.com/NethermindEth/juno/blob/main/.github/workflows/docker-image-build-push.yml) repositories as they both have an implementation of it, and both of them have the condition as well. I thought it was a very good idea to do the check.

Reply:
I wouldn't mind painting CI red in the fork since trying to publish a release on a fork would be incomprehensible. I prefer if such code is introduced by contributors based on their real-world use case, not arbitrarily decided up front just in case. By providing the use case or reasoning it is easy to asses if the change is justified. Otherwise we don't know it the code is useful at all

Having that said it's not a big deal so I'm fine with merging it with the change